lib/Parse/ParseType.cpp

```diff
@@ -180,7 +180,7 @@ ParserResult<TypeRepr> Parser::parseTypeSimple(
   case tok::kw_Self:
   case tok::kw_Any:
   case tok::identifier: {
-    ty = parseTypeIdentifier();
+    ty = parseTypeIdentifier(false, reason);
```

Comment:
If we can use `canParseAsGenericArgumentList()` this would be not needed, but:
```suggestion
    ty = parseTypeIdentifier(/*isParsingQualifiedDeclBaseType=*/false, reason);
```

Reply:
Can you elaborate a little bit more on why do we need to pass the reason here?

Reply:
Yes, this is a good idea